post/index.go

```diff
@@ -0,0 +1,369 @@
+package post
+
+import (
+	"database/sql"
+	"encoding/json"
+	"net/http"
+	"strconv"
+	"utils"
+
+	"github.com/gin-gonic/gin"
+	"github.com/jmoiron/sqlx"
+	log "github.com/sirupsen/logrus"
+)
+
+func New(db *sqlx.DB) *Repository {
+	return &Repository{Db: db}
+}
+
+func (repository *Repository) Get(c *gin.Context) {
+	skip, err := strconv.Atoi(c.DefaultQuery("skip", "0"))
+	if err != nil {
+		log.Warn(err)
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	limit, err := strconv.Atoi(c.DefaultQuery("limit", "10"))
+	if err != nil {
+		log.Warn(err)
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	isResource, err := strconv.ParseBool(c.DefaultQuery("is_resource", "false"))
+	if err != nil {
+		log.Warn(err)
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	isPublished, err := strconv.ParseBool(c.DefaultQuery("is_published", "false"))
+	if err != nil {
+		log.Warn(err)
+		c.AbortWithStatus(http.StatusBadRequest)
+		return
+	}
+
+	var PostData struct {
+		Posts         []Post `json:"posts"`
+		FeaturedPosts []Post `json:"featuredPosts"`
+		Count         int    `json:"count"`
+	}
+
+	new_posts := PostData
+	err, new_posts.Posts = repository.GetPosts(false, isResource, isPublished, limit, skip)
+	if err != nil {
+		c.AbortWithStatus(http.StatusInternalServerError)
+		return
+	}
+
+	if isResource && skip == 0 {
+		err, new_posts.FeaturedPosts = repository.GetPosts(true, isResource, isPublished, 6, 0)
+		if err != nil {
+			c.AbortWithStatus(http.StatusInternalServerError)
+			return
+		}
+	}
+
+	resourceQuery := ""
+	if isResource {
+		resourceQuery = "is_resource = true AND"
+	}
+
+	publishQuery := ""
+	if isPublished {
+		publishQuery = "AND is_published = true"
+	}
+
+	err = repository.Db.Get(&new_posts.Count, `SELECT COUNT(id) FROM posts WHERE `+resourceQuery+` is_featured = false `+publishQuery)
+
+	c.JSON(http.StatusOK, new_posts)
+}
+
+func (repository *Repository) GetPosts(isFeatured, isResource, isPublished bool, limit, skip int) (error, []Post) {
+
+	featuredQuery := ""
+	if isFeatured {
+		featuredQuery = " AND is_featured = true "
+	}
+
+	publishQuery := ""
+	if isPublished {
+		publishQuery = " AND is_published = true "
+	}
+
+	posts := []Post{}
+	err := repository.Db.Select(&posts, `SELECT id, title, slug, is_featured, is_published, published_on, published_at, reading_time, summary  
+										FROM posts 
+										WHERE is_resource = $1`+publishQuery+featuredQuery+`
+										ORDER BY published_on DESC
+										LIMIT $2 OFFSET $3`, isResource, limit, skip)
+	if err != nil {
+		log.Error(err)
+		return err, nil
+	}
+
+	new_posts := []Post{}
+	for _, post := range posts {
+
+		err = repository.Db.Get(&post.Image, `SELECT f.id, f.name, f.alternative_text, f.caption, f.width, f.height, f.formats as format, f.hash, f.ext, f.mime, 
```

Comment:
Instead of querying in loop, the optimized way would be retrieving all the posts' images and then compute the result in loop.